src/SIL.Harmony/Db/CrdtRepository.cs

```diff
@@ -9,24 +9,50 @@
 
 namespace SIL.Harmony.Db;
 
-internal class CrdtRepository(ICrdtDbContext _dbContext, IOptions<CrdtConfig> crdtConfig, DateTimeOffset? ignoreChangesAfter = null)
+internal class CrdtRepository(ICrdtDbContext _dbContext, IOptions<CrdtConfig> crdtConfig,
+    Commit? ignoreChangesAfter = null
+    // DateTimeOffset? ignoreChangesAfter = null
+)
 {
     private IQueryable<ObjectSnapshot> Snapshots => _dbContext.Snapshots.AsNoTracking();
+
+    private IQueryable<ObjectSnapshot> SnapshotsFiltered
+    {
+        get
+        {
+            if (ignoreChangesAfter is not null)
+            {
+                return Snapshots.WhereBefore(ignoreChangesAfter, inclusive: true);
+            }
+            return Snapshots;
+        }
+    }
+    private IQueryable<Commit> Commits
+    {
+        get
+        {
+            if (ignoreChangesAfter is not null)
+            {
+                return _dbContext.Commits.WhereBefore(ignoreChangesAfter, inclusive: true);
+            }
+            return _dbContext.Commits;
+        }
+    }
```

Comment:
A few ideas:
- You still have some occurences of `_dbContext.Snapshots` that could use the `Snapshots` property. Updating those would make it easier to track exactly where we use filtered vs. non-filtered collections
- For new code it would be tricky for contributors to remember when they need to access a filtered copy.
- I'm guessing that the methods that **don't** use filtered collections don't get called when there's a scope/filter configured, so maybe they should throw if there's a scope or maybe you should split this class up (e.g. move the methods that use unfiltered collections into a super class and pass filtered collections from the super class into this class.)

Reply:
Yeah, I've been wrestling with that for a bit, I don't love how it works now as it makes it easy to shoot yourself in the foot. Maybe this is the wrong layer to do it? Maybe I should wrap the db context?

Reply:
Alright I've refactored it so devs don't need to think about scoped vs non scoped in 476f128